src/dotnet/Chat.UI.Blazor/Components/DateVisor/date-visor.ts

```diff
@@ -0,0 +1,45 @@
+import { fromEvent, Subject, takeUntil } from 'rxjs';
+
+export class DateVisor {
+    private readonly dateVisor: HTMLElement;
+    private readonly chatView: HTMLElement;
+    private timer: NodeJS.Timer;
+    private disposed$: Subject<void> = new Subject<void>();
+
+    static create(dateVisor: HTMLElement): DateVisor {
+        return new DateVisor(dateVisor);
+    }
+
+    constructor(dateVisor: HTMLElement) {
+        this.dateVisor = dateVisor;
+        this.chatView = document.body.querySelector('.chat-view');
+
+        if (this.chatView == null)
+            return;
+
+        this.timer = null;
+        fromEvent(this.chatView, 'scroll')
+            .pipe(takeUntil(this.disposed$))
+            .subscribe(this.onScrollHandler);
+    }
+
+    public dispose() {
+        if (this.disposed$.isStopped)
+            return;
+
+        this.disposed$.next();
+        this.disposed$.complete();
+    }
+
+    private onScrollHandler = () => {
```

Comment:
We have `throttle` and `debounce` methods that we use at the `virtual-list` for similar purposes - torn on somthing during scroll and turn off after. Those methods provide much more efficient way to handle timers

Reply:
Done